src/kakarot/instructions/system_operations.cairo

```diff
@@ -1205,9 +1205,13 @@ namespace CreateHelper {
             return evm;
         }
 
-        // Write bytecode and valid jumpdests to Account
+        // Write bytecode, valid jumpdests and code_hash to Account
         let account = State.get_account(evm.message.address.evm);
         let account = Account.set_code(account, evm.return_data_len, evm.return_data);
+        Account.compute_code_hash(account);
+        let (ap_val) = get_ap();
+        let code_hash = cast(ap_val - 1, Uint256*);
+        let account = Account.set_code_hash(account, code_hash);
```

Comment:
we could consider computing the code_hash when we set the code in `Account.set_code` instead of creating another `Account.set_code_hash` indeed, because it creates the risk to have a discrepency between code and code_hash

Reply:
Done should be better